xagon-react-app/src/components/GameComponent/inputManager/FlipGesture.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  AbstractMesh,
+  PointerInfo,
+  Scene,
+  Nullable,
+  StandardMaterial,
+  Vector3,
+} from '@babylonjs/core';
+import { getAssetMesh } from 'utils/scene';
+import Gesture from './Gesture';
+
+interface GestureContext {
+  scene: Scene;
+  triangleMesh: AbstractMesh;
+  scalingRatio: number;
+}
+
+class FlipGesture extends Gesture {
+  private context: GestureContext;
+
+  private firstTriangle: Nullable<AbstractMesh>;
+
+  private secondTriangle: Nullable<AbstractMesh>;
+
+  public constructor(context: GestureContext) {
+    super();
+    this.context = context;
+    this.firstTriangle = null;
+    this.secondTriangle = null;
+  }
+
+  public computeObjSpaceData(assetMesh: AbstractMesh):
+    | {
+        vertices: Vector3[];
+        edges: Vector3[];
+      }
+    | undefined {
+    if (assetMesh && assetMesh.skeleton) {
```

Comment:
Ricordiamoci quando aggiungiamo la classe TriangleMesh che questa funzione potrebbe stare li dentro, cosi come in vertici e gli edge in obj space